### PR TITLE
Skips CLI version checks for unsupported Angular versions.

### PR DESCRIPTION
**Pull request by dgp1130**
By default, the Angular CLI will check if it is out-of-date and download the latest version before actually running the `ng update` command. This means that `npx @angular/cli@8 update ...` is effecitvely equivalent to `npx @angular/cli@latest update ...`. However, only Angular 10+ is currently officially supported, so there is no guaratee that CLI v13 will successfully update from v7 -> v8, and if a user filed a bug it would be closed as unsupported. Instead, the best option for users on unsupported versions is to disable the update check and actually use the destination CLI version, which at least supported the update at the time, even if it isn't supported now.

`NG_DISABLE_VERSION_CHECK=1` will skip the auto-update, so it has been added to each update command. Unfortunately the syntax for Windows is different, so I introduced a new Windows checkbox to show a different command. This required some minor changes to work as expected. The checkbox defaults to the user's platform per `navigator.platform`. Although the values are not strictly defined, "Windows" and "Win32" seem to be sufficient (assuming there are no 16-bit supported Windows devices). See: https://stackoverflow.com/a/19883965.

The Windows commands look a bit awkward with nested, unsecapted quotes, however that [does appear to be the way it is supposed to work here](https://superuser.com/a/950944) and I tested the command on my own Windows box to verify.

### CODE REVIEW

**Comment by dgp1130** [comment]
FYI, we're working to update supported CLI versions to do this automatically, so instead of updating itself to `@angular/cli@latest`, it would update itself to `@angular/cli@${destination_major}.${latest}.${latest}`. This way `ng update` will "just work" for all versions, even when they fall out of support.

For stuff that is already unsupported, we can't easily backport that change, so this PR should clear up the documentation there. Going forward however, `ng update` should continue to work for v10+ even when they become unsupported.

**Comment by dgp1130** [comment]
@mgechev any concerns with this?

File: src/app/app.component.ts [hunk 1]
```diff
@@ -140,9 +146,12 @@ export class AppComponent implements OnInit {
         // Or when the user has a matching option selected
         let skip = false;
         for (const option of this.optionList) {
-          if (step[option.id] && !this.options[option.id]) {
-            skip = true;
-          }
+          // Skip steps which require an option not set by the user.
```
Comment (mgechev) [inline src/app/app.component.ts:149]: Can we keep the braces here and below?

**Review by mgechev — APPROVED** [review]
Just a single nit. LGTM

**Pull request merged**